### ISSUE TITLE
Simple echo/prompt demo for browser-only terminal?

### ISSUE THREAD

**Issue by billyzkid** [sourcelair/xterm.js#515]
The existing demos I've seen demonstrate socket/pty/node integration with the xterm Terminal control, but what if I have a much simpler use case.

I'd like to see a simple browser "echo" demo. I see the existing demo alludes to a "fake" terminal which displays a prompt and writes key input, but this does not show me how to read the user's input (especially when the input spans multiple lines), wire up a standard keyboard shortcut, etc. Also, paste and backspace across multi-line input does not seem obviously supported in this scenario.

**Comment by vincentwoo** [comment]
To piggyback off this, I also think that the latest release should have a simple demo as described above available at http://xtermjs.org/

**Comment by Tyriar** [comment]
Seems like it would be a good improvement to add an input listener to the quick start :+1:

**Comment by parisk** [comment]
Good idea. I'll open up a branch in https://github.com/xtermjs/xtermjs.org.

**Comment by BenHall** [comment]
Katacoda is available for you if it's useful? Would users to deploy the node sample app without having to configure anything...

**Comment by parisk** [comment]
Thanks @BenHall for offering! Can you get into a little bit more details about what would this Node app do?

What might be useful would be if each visitor could get an actual Bash terminal connected to the front-end, but this is quite the opposite of the current issue.

**Comment by vincentwoo** [comment]
We at CoderPad might be willing to sponsor this demo in the future! We run a shit-ton of containers constantly, anyhow.

**Comment by InDieTasten** [comment]
Is this completed? Where can I find it?

@parisk I cannot find a branch besides master publicly available on GitHub

**Comment by DRSDavidSoft** [comment]
Subscribed for updates regarding this

**Comment by LucianBuzzo** [comment]
We're using a simple client side javascript REPL to demo our `Terminal` component (which wraps xterm) in rendition, you can see it in action here https://resin-io-modules.github.io/rendition/?selectedKind=Terminal&selectedStory=Standard&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel
Some of the source code could be leveraged to build a simple demo

**Comment by santosh0705** [comment]
@LucianBuzzo could you share some more information or some demonstration code.

**Comment by LucianBuzzo** [comment]
@santosh0705 Here is the source code for a simple JS repl tethered to an xterm.js instance https://github.com/resin-io-modules/rendition/blob/master/src/stories/Terminal.js#L44
In the Rendition library, we add a simple wrapper around the xterm.js instance and expose it as `this.tty`.

**Comment by wavesoft** [comment]
If you want a more elaborate local-echo controller, you could use this : https://github.com/wavesoft/local-echo
